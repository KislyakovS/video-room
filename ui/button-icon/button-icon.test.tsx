import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import ButtonIcon from './button-icon';

interface MockProps {
    type: string;
}

jest.mock('../icon', () => ({ type }: MockProps) => `Svg Icon - ${type}`);

describe('Component ButtonIcon:', () => {
    it('snapshot', () => {
        const { container } = render(<ButtonIcon icon='search' aria-label='Search' />);

        expect(container).toMatchSnapshot();
    });

    it('should call for a click on the button', async () => {
        const user = userEvent.setup();
        const onClick = jest.fn();

        const { getByRole } = render(<ButtonIcon onClick={onClick} icon='search' aria-label='Search' />);
        const button = getByRole('button');

        await user.click(button);

        expect(onClick).toHaveBeenCalled();
    });
});