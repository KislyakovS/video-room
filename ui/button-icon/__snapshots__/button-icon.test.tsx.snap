// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ButtonIcon: snapshot 1`] = `
<div>
  <button
    aria-label="Search"
    class="button"
  >
    Svg Icon - search
  </button>
</div>
`;
